src/intrinsics.ts

```diff
@@ -0,0 +1,158 @@
+    'SyntaxError',
+    'TypeError',
+    'Uint8Array',
+    'Uint8ClampedArray',
+    'Uint16Array',
+    'Uint32Array',
+    'URIError',
+    'WeakMap',
+    'WeakSet',
+
+    // *** 18.4 Other Properties of the Global Object
+    'Atomics',
+    'JSON',
+    'Math',
+    'Reflect',
+
+    // *** Annex B
+    'escape',
+    'unescape',
+
+    // *** ECMA-402
+    'Intl', // Unstable
+]);
+
+// These are foundational things that should never be wrapped but are equivalent
+// TODO: revisit this list.
+const ReflectiveIntrinsicObjectNames = [
+    'Array',
+    'Error',
+    'EvalError',
+    'Function',
+    'Object',
+    'RangeError',
+    'ReferenceError',
+    'SyntaxError',
+    'TypeError',
+    'URIError',
+];
+
+function getReflectiveIntrinsics(global: typeof globalThis): ReflectiveIntrinsicsMap {
+    let reflectiveIntrinsics: ReflectiveIntrinsicsMap | undefined = WeakMapGet(cachedReflectiveIntrinsicsMap, global);
+    if (!isUndefined(reflectiveIntrinsics)) {
+        return reflectiveIntrinsics;
+    }
+    reflectiveIntrinsics = ObjectCreate(null) as ReflectiveIntrinsicsMap;
+    WeakMapSet(cachedReflectiveIntrinsicsMap, global, reflectiveIntrinsics);
+    // remapping intrinsics that are realm's agnostic
+    for (let i = 0, len = ReflectiveIntrinsicObjectNames.length; i < len; i += 1) {
+        const name = ReflectiveIntrinsicObjectNames[i];
+        reflectiveIntrinsics[name] = global[name];
+    }
+    return reflectiveIntrinsics;
+}
+
+export function linkIntrinsics(
+    env: SecureEnvironment,
+    blueGlobalThis: typeof globalThis,
+    redGlobalThis: typeof globalThis
+) {
+    // remapping intrinsics that are realm's agnostic
+    const blueIntrinsics = getReflectiveIntrinsics(blueGlobalThis);
+    const redIntrinsics = getReflectiveIntrinsics(redGlobalThis);
+    for (let i = 0, len = ReflectiveIntrinsicObjectNames.length; i < len; i += 1) {
+        const name = ReflectiveIntrinsicObjectNames[i];
+        const blue = blueIntrinsics[name];
+        const red = redIntrinsics[name];
+        env.setRefMapEntries(red, blue);
+        env.setRefMapEntries(red.prototype, blue.prototype);
+    }
+}
+
+export function getFilteredEndowmentDescriptors(endowments: object): PropertyDescriptorMap {
+    const to: PropertyDescriptorMap = ObjectCreate(null);
+    const endowmentsDescriptors = getOwnPropertyDescriptors(endowments);
+    const globalKeys = ownKeys(endowmentsDescriptors);
+    for (let i = 0, len = globalKeys.length; i < len; i++) {
+        // forcing to string here because of TypeScript's PropertyDescriptorMap definition, which doesn't
+        // support symbols as entries.
+        const key = globalKeys[i] as string;
+        // avoid overriding ECMAScript global names that correspond
+        // to global intrinsics. This guarantee that those entries
+        // will be ignored if present in the endowments object.
+        // TODO: what if the intent is to polyfill one of those
+        // intrinsics?
```

Comment:
in the next branch, we only prevent endowments to redefine global names that can give you access to an undeniable element. This one just avoid endowments that match any of the global intrinsic names all together. That means a endowment for `JSON` will fail to be applied, and will be just ignored.